### DISCUSSION TITLE
Task runner vs scripts

### DISCUSSION THREAD

**Discussion by bryantbiggs** [lalrpop/lalrpop#730] [General]
Theres various scripts and a little bit of duplication between CI and scripts. I'm happy to take on making this a bit better - something that can be used both locally by developers as well as by the CI checks (so they both are doing the same checks the same way). Before I started though, I wanted to see if there was any preference on tooling or approach. A couple options, please suggest others:
- Standard Makefile (GNUMakefile)
- [cargo-make](https://sagiegurari.github.io/cargo-make/)
- [justfile](https://github.com/casey/just)

**Comment by bryantbiggs** [comment]
We could also look at the release process in a 2nd step as well - see if there is something that can be done to simplify the code generation to ensure it happens at the appropriate time, bumping the version, etc.

**Comment by bryantbiggs** [comment]
Makefile seemed like the least amount of overhead (should most likely already be present on most machines) so I took an initial pass at this https://github.com/lalrpop/lalrpop/pull/731

**Comment by yannham** [comment]
Well, I would suggest [Nix](https://nixos.org/), as a believer :stuck_out_tongue: , but mostly as a half-joke: I don't think it would pay for the complexity in such a project. FWIW, starting with something simple and ubiquitous like Makefile and see if we need something more capable (or more manageable, if the complexity grows) later sounds like a sensible approach to me :+1:
